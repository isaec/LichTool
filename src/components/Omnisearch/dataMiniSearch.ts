import { spellArray } from "@src/dataLookup";
import MiniSearch from "minisearch";

// use the same minisearch for each search instance
const searchEngine = new MiniSearch({
  fields: ["name"],
  searchOptions: {
    fuzzy: 0.5,
    prefix: true,
    weights: {
      fuzzy: 0.65,
      prefix: 1,
    },
  },
});
searchEngine.addAll(spellArray);

export default searchEngine;
