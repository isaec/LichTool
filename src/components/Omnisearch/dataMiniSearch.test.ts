import { describe, it, expect } from "vitest";
import dataMiniSearch from "./dataMiniSearch";

describe("dataMiniSearch", () => {
  it.each(["fireball", "fire blt", "antipathy sympathy", "mine black", "find"])(
    "matches snapshot for search results with query %s",
    (query) => {
      expect(dataMiniSearch.search(query)).toMatchSnapshot();
    }
  );
});
