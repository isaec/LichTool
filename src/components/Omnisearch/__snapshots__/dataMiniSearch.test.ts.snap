// Vitest Snapshot v1

exports[`dataMiniSearch > matches snapshot for search results with query antipathy sympathy 1`] = `
[
  {
    "id": "spell_PHB-Antipathy/Sympathy",
    "terms": [
      "antipathy",
      "sympathy",
    ],
  },
  {
    "id": "spell_PHB-Find-the-Path",
    "terms": [
      "path",
    ],
  },
  {
    "id": "spell_PHB-Antimagic-Field",
    "terms": [
      "antimagic",
    ],
  },
  {
    "id": "action_PHB-Other-Activity",
    "terms": [
      "activity",
    ],
  },
  {
    "id": "spell_PHB-Antilife-Shell",
    "terms": [
      "antilife",
    ],
  },
  {
    "id": "action_DMG-Activate-an-Item",
    "terms": [
      "activate",
    ],
  },
  {
    "id": "spell_PHB-Enhance-Ability",
    "terms": [
      "ability",
    ],
  },
  {
    "id": "spell_PHB-Animate-Dead",
    "terms": [
      "animate",
    ],
  },
  {
    "id": "spell_PHB-Animate-Objects",
    "terms": [
      "animate",
    ],
  },
  {
    "id": "spell_PHB-Conjure-Animals",
    "terms": [
      "animals",
    ],
  },
  {
    "id": "spell_PHB-Animal-Friendship",
    "terms": [
      "animal",
    ],
  },
  {
    "id": "spell_PHB-Animal-Messenger",
    "terms": [
      "animal",
    ],
  },
  {
    "id": "spell_PHB-Animal-Shapes",
    "terms": [
      "animal",
    ],
  },
  {
    "id": "spell_PHB-Speak-with-Animals",
    "terms": [
      "animals",
    ],
  },
  {
    "id": "spell_PHB-Locate-Animals-or-Plants",
    "terms": [
      "animals",
    ],
  },
]
`;

exports[`dataMiniSearch > matches snapshot for search results with query find 1`] = `
[
  {
    "id": "spell_PHB-Find-Familiar",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Steed",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Traps",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-the-Path",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Create-Food-and-Water",
    "terms": [
      "and",
      "food",
    ],
  },
  {
    "id": "spell_PHB-Purify-Food-and-Drink",
    "terms": [
      "and",
      "food",
    ],
  },
  {
    "id": "spell_PHB-Mind-Blank",
    "terms": [
      "mind",
    ],
  },
  {
    "id": "item_PHB-Wand",
    "terms": [
      "wand",
    ],
  },
  {
    "id": "spell_PHB-Antimagic-Field",
    "terms": [
      "field",
    ],
  },
  {
    "id": "spell_PHB-Aid",
    "terms": [
      "aid",
    ],
  },
  {
    "id": "spell_PHB-Wind-Walk",
    "terms": [
      "wind",
    ],
  },
  {
    "id": "spell_PHB-Wind-Wall",
    "terms": [
      "wind",
    ],
  },
  {
    "id": "spell_PHB-Warding-Bond",
    "terms": [
      "bond",
    ],
  },
  {
    "id": "item_PHB-Ring-Mail",
    "terms": [
      "ring",
    ],
  },
  {
    "id": "item_PHB-Hand-Crossbow",
    "terms": [
      "hand",
    ],
  },
  {
    "id": "spell_PHB-Mage-Hand",
    "terms": [
      "hand",
    ],
  },
  {
    "id": "spell_PHB-Gust-of-Wind",
    "terms": [
      "wind",
    ],
  },
  {
    "id": "action_PHB-End-Concentration",
    "terms": [
      "end",
    ],
  },
  {
    "id": "spell_PHB-Faerie-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Bolt",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Shield",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Storm",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Guards-and-Wards",
    "terms": [
      "and",
    ],
  },
  {
    "id": "spell_PHB-Detect-Evil-and-Good",
    "terms": [
      "and",
    ],
  },
  {
    "id": "spell_PHB-Detect-Poison-and-Disease",
    "terms": [
      "and",
    ],
  },
  {
    "id": "spell_PHB-Dispel-Evil-and-Good",
    "terms": [
      "and",
    ],
  },
  {
    "id": "spell_PHB-Protection-from-Evil-and-Good",
    "terms": [
      "and",
    ],
  },
]
`;

exports[`dataMiniSearch > matches snapshot for search results with query fire blt 1`] = `
[
  {
    "id": "spell_PHB-Fire-Bolt",
    "terms": [
      "fire",
      "bolt",
    ],
  },
  {
    "id": "spell_PHB-Delayed-Blast-Fireball",
    "terms": [
      "fireball",
      "blast",
    ],
  },
  {
    "id": "spell_PHB-Fireball",
    "terms": [
      "fireball",
    ],
  },
  {
    "id": "spell_PHB-Faerie-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Shield",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Storm",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "item_PHB-Pike",
    "terms": [
      "pike",
    ],
  },
  {
    "id": "spell_PHB-Blur",
    "terms": [
      "blur",
    ],
  },
  {
    "id": "action_PHB-Hide",
    "terms": [
      "hide",
    ],
  },
  {
    "id": "spell_PHB-Fly",
    "terms": [
      "fly",
    ],
  },
  {
    "id": "item_PHB-Net",
    "terms": [
      "net",
    ],
  },
  {
    "id": "spell_PHB-Gaseous-Form",
    "terms": [
      "form",
    ],
  },
  {
    "id": "spell_PHB-False-Life",
    "terms": [
      "life",
    ],
  },
  {
    "id": "spell_PHB-Legend-Lore",
    "terms": [
      "lore",
    ],
  },
  {
    "id": "spell_PHB-Time-Stop",
    "terms": [
      "time",
    ],
  },
  {
    "id": "spell_PHB-Eldritch-Blast",
    "terms": [
      "blast",
    ],
  },
  {
    "id": "item_PHB-Crossbow-Bolt",
    "terms": [
      "bolt",
    ],
  },
  {
    "id": "spell_PHB-Guiding-Bolt",
    "terms": [
      "bolt",
    ],
  },
  {
    "id": "spell_PHB-Lightning-Bolt",
    "terms": [
      "bolt",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Force",
    "terms": [
      "force",
    ],
  },
  {
    "id": "item_PHB-Hide-Armor",
    "terms": [
      "hide",
    ],
  },
  {
    "id": "spell_PHB-Cure-Wounds",
    "terms": [
      "cure",
    ],
  },
  {
    "id": "disease_DMG-Sight-Rot",
    "terms": [
      "rot",
    ],
  },
  {
    "id": "item_PHB-Crossbow-Bolts-20",
    "terms": [
      "bolts",
    ],
  },
  {
    "id": "spell_PHB-Ice-Storm",
    "terms": [
      "ice",
    ],
  },
  {
    "id": "spell_PHB-Mass-Cure-Wounds",
    "terms": [
      "cure",
    ],
  },
  {
    "id": "spell_PHB-Find-Familiar",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Steed",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Traps",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Ice",
    "terms": [
      "ice",
    ],
  },
  {
    "id": "spell_PHB-Find-the-Path",
    "terms": [
      "find",
    ],
  },
]
`;

exports[`dataMiniSearch > matches snapshot for search results with query fireball 1`] = `
[
  {
    "id": "spell_PHB-Fireball",
    "terms": [
      "fireball",
    ],
  },
  {
    "id": "spell_PHB-Delayed-Blast-Fireball",
    "terms": [
      "fireball",
    ],
  },
  {
    "id": "spell_PHB-Word-of-Recall",
    "terms": [
      "recall",
    ],
  },
  {
    "id": "spell_PHB-Antimagic-Field",
    "terms": [
      "field",
    ],
  },
  {
    "id": "spell_PHB-Feather-Fall",
    "terms": [
      "fall",
    ],
  },
  {
    "id": "spell_PHB-Faerie-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Bolt",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Shield",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Storm",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Fire",
    "terms": [
      "fire",
    ],
  },
]
`;

exports[`dataMiniSearch > matches snapshot for search results with query mine black 1`] = `
[
  {
    "id": "spell_PHB-Mind-Blank",
    "terms": [
      "mind",
      "blank",
    ],
  },
  {
    "id": "spell_PHB-Bane",
    "terms": [
      "bane",
    ],
  },
  {
    "id": "item_PHB-Mace",
    "terms": [
      "mace",
    ],
  },
  {
    "id": "spell_PHB-Wind-Walk",
    "terms": [
      "wind",
      "walk",
    ],
  },
  {
    "id": "spell_PHB-Flame-Blade",
    "terms": [
      "flame",
      "blade",
    ],
  },
  {
    "id": "spell_PHB-Blink",
    "terms": [
      "blink",
    ],
  },
  {
    "id": "spell_PHB-Maze",
    "terms": [
      "maze",
    ],
  },
  {
    "id": "item_PHB-Pike",
    "terms": [
      "pike",
    ],
  },
  {
    "id": "spell_PHB-Alarm",
    "terms": [
      "alarm",
    ],
  },
  {
    "id": "item_PHB-Flail",
    "terms": [
      "flail",
    ],
  },
  {
    "id": "spell_PHB-Bless",
    "terms": [
      "bless",
    ],
  },
  {
    "id": "item_PHB-Lance",
    "terms": [
      "lance",
    ],
  },
  {
    "id": "spell_PHB-Knock",
    "terms": [
      "knock",
    ],
  },
  {
    "id": "spell_PHB-Branding-Smite",
    "terms": [
      "smite",
    ],
  },
  {
    "id": "action_PHB-Hide",
    "terms": [
      "hide",
    ],
  },
  {
    "id": "action_PHB-Attack",
    "terms": [
      "attack",
    ],
  },
  {
    "id": "spell_PHB-Blur",
    "terms": [
      "blur",
    ],
  },
  {
    "id": "spell_PHB-False-Life",
    "terms": [
      "life",
    ],
  },
  {
    "id": "spell_PHB-Move-Earth",
    "terms": [
      "move",
    ],
  },
  {
    "id": "spell_PHB-Time-Stop",
    "terms": [
      "time",
    ],
  },
  {
    "id": "item_PHB-Ring-Mail",
    "terms": [
      "ring",
    ],
  },
  {
    "id": "spell_PHB-Arcane-Lock",
    "terms": [
      "lock",
    ],
  },
  {
    "id": "spell_PHB-Minor-Illusion",
    "terms": [
      "minor",
    ],
  },
  {
    "id": "spell_PHB-Blade-Barrier",
    "terms": [
      "blade",
    ],
  },
  {
    "id": "spell_PHB-Eldritch-Blast",
    "terms": [
      "blast",
    ],
  },
  {
    "id": "spell_PHB-Dominate-Beast",
    "terms": [
      "beast",
    ],
  },
  {
    "id": "spell_PHB-Plant-Growth",
    "terms": [
      "plant",
    ],
  },
  {
    "id": "spell_PHB-Rope-Trick",
    "terms": [
      "trick",
    ],
  },
  {
    "id": "item_PHB-Hide-Armor",
    "terms": [
      "hide",
    ],
  },
  {
    "id": "spell_PHB-Mage-Armor",
    "terms": [
      "mage",
    ],
  },
  {
    "id": "spell_PHB-Mage-Hand",
    "terms": [
      "mage",
    ],
  },
  {
    "id": "action_PHB-Opportunity-Attack",
    "terms": [
      "attack",
    ],
  },
  {
    "id": "spell_PHB-Cone-of-Cold",
    "terms": [
      "cone",
    ],
  },
  {
    "id": "spell_PHB-Zone-of-Truth",
    "terms": [
      "zone",
    ],
  },
  {
    "id": "spell_PHB-Beacon-of-Hope",
    "terms": [
      "beacon",
    ],
  },
  {
    "id": "spell_PHB-Conjure-Minor-Elementals",
    "terms": [
      "minor",
    ],
  },
  {
    "id": "spell_PHB-Delayed-Blast-Fireball",
    "terms": [
      "blast",
    ],
  },
  {
    "id": "item_PHB-War-Pick",
    "terms": [
      "pick",
    ],
  },
  {
    "id": "item_PHB-Plate-Armor",
    "terms": [
      "plate",
    ],
  },
  {
    "id": "spell_PHB-Plane-Shift",
    "terms": [
      "plane",
    ],
  },
  {
    "id": "spell_PHB-Wind-Wall",
    "terms": [
      "wind",
    ],
  },
  {
    "id": "spell_PHB-Pass-without-Trace",
    "terms": [
      "trace",
    ],
  },
  {
    "id": "spell_PHB-Ice-Storm",
    "terms": [
      "ice",
    ],
  },
  {
    "id": "spell_PHB-Find-Familiar",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Steed",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Find-Traps",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Water-Walk",
    "terms": [
      "walk",
    ],
  },
  {
    "id": "spell_PHB-Faerie-Fire",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Bolt",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Shield",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Fire-Storm",
    "terms": [
      "fire",
    ],
  },
  {
    "id": "spell_PHB-Hunter's-Mark",
    "terms": [
      "mark",
    ],
  },
  {
    "id": "item_PHB-Half-Plate-Armor",
    "terms": [
      "plate",
    ],
  },
  {
    "id": "spell_PHB-Contact-Other-Plane",
    "terms": [
      "plane",
    ],
  },
  {
    "id": "spell_PHB-Gust-of-Wind",
    "terms": [
      "wind",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Ice",
    "terms": [
      "ice",
    ],
  },
  {
    "id": "spell_PHB-Continual-Flame",
    "terms": [
      "flame",
    ],
  },
  {
    "id": "spell_PHB-Flame-Strike",
    "terms": [
      "flame",
    ],
  },
  {
    "id": "spell_PHB-Produce-Flame",
    "terms": [
      "flame",
    ],
  },
  {
    "id": "spell_PHB-Sacred-Flame",
    "terms": [
      "flame",
    ],
  },
  {
    "id": "spell_PHB-Find-the-Path",
    "terms": [
      "find",
    ],
  },
  {
    "id": "spell_PHB-Wall-of-Fire",
    "terms": [
      "fire",
    ],
  },
]
`;
